src/bytecode/mod.rs

```diff
@@ -0,0 +1,51 @@
+pub mod instructions;
+
+use std::vec::Vec;
+use std::fmt;
+use self::instructions::Instr;
+
+/// A simpler representation of Lua
+pub struct LuaBytecode {
+    block: Vec<Instr>,
+    ids: u64
+}
+
+impl<'a> LuaBytecode {
+    pub fn new() -> LuaBytecode {
+        LuaBytecode {
+            block: vec![],
+            ids: 0
+        }
+    }
+
+    /// Get the list of instructions that can be executed in order
+    /// to perform some computation.
+    pub fn get_instrs(&'a self) -> &'a Vec<Instr> {
```

Comment:
In general, I'd suggest that you probably don't want a method like this, because it effectively locks the `struct` into one representation forever. If you have a method called something like `get_instr(x: usize) -> Instr` which loads one instruction at a time (or, if you need a range, an iterator), you keep your options for the future more open.

Reply:
I see, that makes sense. I will have to rethink this struct anyway, because I will have to consider functions, blocks, etc. and the current representation is limited to one block.